src/libs/Authentication.ts

```diff
@@ -62,48 +62,55 @@ function reauthenticate(command = ''): Promise<void> {
+            }
 
-        // If we reauthenticated due to an expired delegate token, restore the delegate's original account.
-        // This is because the credentials used to reauthenticate were for the delegate's original account, and not for the account they were connected as.
-        if (Delegate.isConnectedAsDelegate()) {
-            Log.info('Reauthenticated while connected as a delegate. Restoring original account.');
-            Delegate.restoreDelegateSession(response);
-            return;
-        }
+            // If we reauthenticated due to an expired delegate token, restore the delegate's original account.
+            // This is because the credentials used to reauthenticate were for the delegate's original account, and not for the account they were connected as.
+            if (Delegate.isConnectedAsDelegate()) {
+                Log.info('Reauthenticated while connected as a delegate. Restoring original account.');
+                Delegate.restoreDelegateSession(response);
+                return;
+            }
 
-        // Update authToken in Onyx and in our local variables so that API requests will use the new authToken
-        updateSessionAuthTokens(response.authToken, response.encryptedAuthToken);
+            // Update authToken in Onyx and in our local variables so that API requests will use the new authToken
+            updateSessionAuthTokens(response.authToken, response.encryptedAuthToken);
 
-        // Note: It is important to manually set the authToken that is in the store here since any requests that are hooked into
-        // reauthenticate .then() will immediate post and use the local authToken. Onyx updates subscribers lately so it is not
-        // enough to do the updateSessionAuthTokens() call above.
-        NetworkStore.setAuthToken(response.authToken ?? null);
+            // Note: It is important to manually set the authToken that is in the store here since any requests that are hooked into
+            // reauthenticate .then() will immediate post and use the local authToken. Onyx updates subscribers lately so it is not
+            // enough to do the updateSessionAuthTokens() call above.
+            NetworkStore.setAuthToken(response.authToken ?? null);
 
-        // The authentication process is finished so the network can be unpaused to continue processing requests
-        NetworkStore.setIsAuthenticating(false);
-    });
+            // The authentication process is finished so the network can be unpaused to continue processing requests
+            NetworkStore.setIsAuthenticating(false);
+        })
+        .catch((error) => {
+            // In case the atuhentication call throws error, we need to sign user out as most likely they are missing credentials
+            NetworkStore.setIsAuthenticating(false);
+            Log.hmmm('Redirecting to Sign In because we failed to reauthenticate', {error});
+            redirectToSignIn('passwordForm.error.fallback');
+        });
```

Comment:
If the Authenticate call throws error due to missing params, we just entered infinite loop of failing requests as we never catch it and sign user out